src/vivarium/framework/engine.py

```diff
@@ -349,6 +350,26 @@ def get_population(self, untracked: bool = True) -> pd.DataFrame:
     def __repr__(self):
         return f"SimulationContext({self.name})"
 
+    def get_number_of_steps(self, start_time: pd.Timestamp) -> int:
```

Comment:
This method shouldn't take any arguments. It should be able to figure out the start time on its own.